### PR TITLE
Add TO API v5

### PR DESCRIPTION
**Pull request by ocket8888**
Adds version 5.0 of the Traffic Ops API.

Everything still uses v4 (which doesn't matter yet as there are no differences between v4 and v5 as of this PR)

<hr/>

## Which Traffic Control components are affected by this PR?
- Documentation
- Traffic Control Clients (Python, TO Go client (adds new))
- Traffic Ops
- Automation (GitHub Actions)

## What is the best way to verify this PR?
Make sure all of the new APIv5 TO Go client integration tests pass, make sure the docs build without errors or warnings and are complete and accurate.

## PR submission checklist
- [x] This PR has tests
- [x] This PR has documentation
- [x] This PR has a CHANGELOG.md entry
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY**

### CODE REVIEW

File: traffic_ops/testing/api/utils/utils.go [hunk 1]
```diff
@@ -98,13 +99,11 @@ func CreateV4Session(t *testing.T, TrafficOpsURL string, username string, passwo
...
 
-// V3TestCase is the type of the V3TestData struct.
-// Uses nested map to represent the method being tested and the test's description.
-type V3TestCase map[string]map[string]V3TestData
-
-// V4TestCase is the type of the V4TestData struct.
-// Uses nested map to represent the method being tested and the test's description.
-type V4TestCase map[string]map[string]V4TestData
+func CreateV5Session(t *testing.T, TrafficOpsURL, username, password string, toReqTimeout int) *v5client.Session {
+	userSession, _, err := v5client.LoginWithAgent(TrafficOpsURL, username, password, true, "to-api-v4-client-tests", false, time.Second*time.Duration(toReqTimeout))
```
Comment (srijeet0406) [inline traffic_ops/testing/api/utils/utils.go:103]: Should be "to-api-v5-client-tests"

File: docs/source/api/v4/deliveryserviceserver.rst [hunk 2]
```diff
@@ -58,7 +58,7 @@ Response Structure
 ------------------
 Unlike most API endpoints, this will return a JSON response body containing both a "response" object as well as other, top-level fields (besides the optional "alerts" field). For this reason, this section contains a "response" key, which normally is implicit.
 
-.. seealso:: :ref:`to-api-response-structure`
+.. seealso:: :ref:`to-api-v4-response-structure`
```
[the remark this answers is not in the document]
Comment (ocket8888) [reply docs/source/api/v4/deliveryserviceserver.rst:61]: no, this is a reference to an existing label, not a definition of a new label. Also though, a new label can be defined at any point throughout the document, and need not be at any particular position within any section or source file.

But the label being referenced here doesn't exist. I need to revert this change.

File: traffic_ops/v5-client/README.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
+# Traffic Ops Go Client
+
```
[the remark this answers is not in the document]
Comment (ocket8888) [reply traffic_ops/v5-client/README.md:2]: No, TO APIv5 is not released, so it doesn't need that notice. APIv2 and APIv3 didn't have it, for example.

File: docs/source/api/v4/servers.rst [hunk 4]
```diff
@@ -40,7 +40,7 @@ Request Structure
 	| cachegroupName | no       | Return only those servers within the :term:`Cache Group` that has this :ref:`cache-group-name`                    |
```
[the remark this answers is not in the document]
Comment (ocket8888) [reply docs/source/api/v4/servers.rst:40]: It is, but not on this particular line. Table is malformed a few lines down.

**Review by srijeet0406 — APPROVED** [review]

**Pull request merged**